### PR TITLE
Add gather_out arg to Linear

### PR DESCRIPTION
**Pull request by Wesley-Jzy**
To use gather_out param in Linear1D, expose the interface.

### CODE REVIEW

File: colossalai/nn/layer/colossalai_layer/linear.py [hunk 1]
```diff
@@ -65,15 +66,27 @@ def __init__(self,
...
+            else:
+                self.layer = _parallel_linear[tensor_parallel](
+                    in_features,
+                    out_features,
+                    bias=bias,
+                    dtype=dtype,
+                    weight_initializer=weight_initializer,
+                    bias_initializer=bias_initializer,
+                    **kwargs,
+                )
```
Comment (feifeibear) [inline colossalai/nn/layer/colossalai_layer/linear.py:89]: self.layer = _parallel_linear[tensor_parallel](
                    in_features,
                    out_features,
                    bias=bias,
                    dtype=dtype,
gather_output = tensor_parallel == '1d',
                    weight_initializer=weight_initializer,
                    bias_initializer=bias_initializer,
                    **kwargs,
                )

**Review by feifeibear — APPROVED** [review]

**Review by feifeibear — APPROVED** [review]

File: colossalai/nn/layer/colossalai_layer/linear.py [hunk 2]
```diff
@@ -65,15 +66,27 @@ def __init__(self,
...
-            self.layer = _parallel_linear[tensor_parallel](
-                in_features,
-                out_features,
-                bias=bias,
-                dtype=dtype,
-                weight_initializer=weight_initializer,
-                bias_initializer=bias_initializer,
-                **kwargs,
-            )
+            if tensor_parallel == '1d': # gather_out arg is available
```
Comment (kurisusnowdeng) [inline colossalai/nn/layer/colossalai_layer/linear.py:69]: We can actually use `inspect.signature` to check if gather_output is an argument of the specific linear module, for the purpose of avoiding writing the same codes twice.

**Review by kurisusnowdeng — CHANGES_REQUESTED** [review]

File: colossalai/nn/layer/colossalai_layer/linear.py (hunk 2 above)
Comment (Wesley-Jzy) [reply colossalai/nn/layer/colossalai_layer/linear.py:69]: ok，I will change it.

Comment (Wesley-Jzy) [reply colossalai/nn/layer/colossalai_layer/linear.py:69]: I use another way to eliminate double invoke. But I wonder how we can use inspect.signature to invoke function with different params?

Comment (Wesley-Jzy) [reply colossalai/nn/layer/colossalai_layer/linear.py:76]: @kurisusnowdeng Done

**Comment by feifeibear** [comment]
Can not pass CI
image

File: colossalai/nn/layer/colossalai_layer/linear.py (hunk 2 above)
Comment (Wesley-Jzy) [reply colossalai/nn/layer/colossalai_layer/linear.py]: Done. @kurisusnowdeng

**Review by kurisusnowdeng — APPROVED** [review]

**Comment by Wesley-Jzy** [comment]
> Can not pass CI image

Fixed.

**Pull request merged**